### PR TITLE
[TS] LPS-201892

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

File: modules/apps/server/server-admin-web-test/src/testIntegration/java/com/liferay/server/admin/web/internal/portlet/action/test/EditServerMVCActionCommandTest.java [hunk 1]
```diff
@@ -184,6 +197,16 @@ public void testCleanUpOrphanedPortletPreferencesWithProperPortletPreferences()
...
 
+	@Test
+	public void testUpdateMailCompanyPortletPreferences() throws Exception {
+		_testUpdateMailPortletPreferences(_portal.getDefaultCompanyId());
+	}
+
+	@Test
+	public void testUpdateMailSystemPortletPreferences() throws Exception {
+		_testUpdateMailPortletPreferences(0L);
+	}
```
Comment (yuhai) [inline modules/apps/server/server-admin-web-test/src/testIntegration/java/com/liferay/server/admin/web/internal/portlet/action/test/EditServerMVCActionCommandTest.java:208]: @marianoalvarosaiz, please replace **0L** with **CompanyConstants.SYSTEM**.

File: modules/apps/server/server-admin-web-test/src/testIntegration/java/com/liferay/server/admin/web/internal/portlet/action/test/EditServerMVCActionCommandTest.java [hunk 2]
```diff
@@ -209,6 +232,133 @@ private LayoutRevision _getLayoutRevision() throws Exception {
...
+
+			_testUpdateMailPortletPreferences(
+				() -> Boolean.FALSE, () -> 0, () -> StringPool.BLANK,
+				portletPreferences);
+		}
+		finally {
+			for (String key : new String[0]) {
+				portletPreferences.reset(key);
+			}
+		}
```
Comment (yuhai) [inline modules/apps/server/server-admin-web-test/src/testIntegration/java/com/liferay/server/admin/web/internal/portlet/action/test/EditServerMVCActionCommandTest.java:359]: @marianoalvarosaiz, the code `portletPreferences.reset(key);` can't be executed for now.  For the reset part, we may use the below way.

For system setting, it doesn't exist data in the database by default. So we may use PortalPreferencesLocalService to delete the generated portalpreferences data. For example,

```
PortalPreferencesWrapper portalPreferencesWrapper = (PortalPreferencesWrapper)portletPreferences;

PortalPreferencesImpl portalPreferencesImpl = portalPreferencesWrapper.getPortalPreferencesImpl();

PortalPreferences portalPreferences = PortalPreferencesLocalService.fetchPortalPreferences(portalPreferencesImpl.getOwnerId(), portalPreferencesImpl.getOwnerType());

PortalPreferencesLocalService.delete(portalPreferences);
```
For the above code fragment, you may place it to finally blcok in the _testUpdateMailPortletPreferences() 

For the instance setting, we may use new company and then delete the company in the testUpdateMailCompanyPortletPreferences().

**Comment by yuhai** [comment]
@marianoalvarosaiz, please see the above comments.

In addition, after the pr, `mail_fields.jsp` only was invoked by `company_mail.jsp`, `system_mail_fields.jsp` only was invoked by `mail.jsp`. So please inline `mail_fields.jsp` to `company_mail.jsp` and `system_mail_fields.jsp` to `mail.jsp`. I think you may append new commit to do this.

In `system_mail_fields.jsp`, we may remove the form data `<aui:input name="preferencesCompanyId" type="hidden" value="<%= 0L %>" />` because we use `ParamUtil.getLong(actionRequest, "preferencesCompanyId"))` to get it, the default value is 0. 

Please kindly let me know if you have other concerns.

Thanks,
Hai

**Comment by marianoalvarosaiz** [comment]
See https://github.com/liferay-core-infra/liferay-portal/pull/6411

**Pull request closed without merging**